xbmc/network/WebServer.cpp

```diff
@@ -1035,10 +1035,43 @@ void CWebServer::ContentReaderFreeCallback(void *cls)
 #endif
 }
 
+// local helper
+static void MHDPanicHandler(void* unused, const char* file, unsigned int line, const char *reason)
+{
+  CLog::Log(LOGSEVERE, "MHD serious error: reason \"%s\" in file \"%s\" at line %ui", reason ? reason : "",
+            file ? file : "", line);
+  throw exception("MHD serious error"); // FIXME: better solution?
```

Comment:
This looks a bit harsh. Maybe display a kai toast notification that the webserver has died? And/Or add some way to automatically restart it?
IMO the webserver is not critical for the functionality of Kodi (obviously a lot of people use it to control Kodi but not all do) so it shouldn't lead to a crash of the whole application.